ddtrace/contrib/trace_utils.py

```diff
@@ -595,3 +598,66 @@ def set_user(tracer, user_id, name=None, email=None, scope=None, role=None, sess
             "See https://docs.datadoghq.com/security_platform/application_security/setup_and_configure/"
             "?tab=set_user&code-lang=python for more information.",
         )
+
+
+def track_user_login(
+    tracer,
+    user_id,
+    success=True,
+    metadata=None,
+    name=None,
+    email=None,
+    scope=None,
+    role=None,
+    session_id=None,
+    propagate=False,
+):
+    # type: (Tracer, str, bool, Optional[dict], Optional[str], Optional[str], Optional[str], Optional[str], Optional[str], bool) -> None # noqa: E501
+
+    span = tracer.current_root_span()
+    if span:
+        success_str = "success" if success else "failure"
+
+        span.set_tag_str("%s.%s.track" % (APPSEC.USER_LOGIN_EVENT_PREFIX, success_str), "true")
+        # usr.id will be set by set_user
+        if metadata is not None:
+            for k, v in six.iteritems(metadata):
+                span.set_tag_str("%s.%s.%s" % (APPSEC.USER_LOGIN_EVENT_PREFIX, success_str, k), str(v))
+
+        span.set_tag_str(constants.MANUAL_KEEP_KEY, "true")
+
+        if success:
+            set_user(tracer, user_id, name, email, scope, role, session_id, propagate)
+        else:
+            span.set_tag_str("%s.failure.%s" % (APPSEC.USER_LOGIN_EVENT_PREFIX, user.ID), str(user_id))
+    else:
+        log.warning(
```

Comment:
Not sure if this should be logged as warning, I think maybe debug is better

Reply:
The spec says a warning should be logged.

Reply:
IMO, worth something like warning or error so that the user sees it as soon as possible, because this will be likely due to a usage mistake I think.